### PR TITLE
Issue 33 password leak

### LINKED ISSUE #33: Server password leaks into logs
**Describe the bug**
When the client is validating the configuration it may throw an exception that prints the configuration to the logs:

```Caused by: clojure.lang.ExceptionInfo: setting maxCommandsInProcess>0 and maxCommandsInQueue=0 creates an unbounded delay queue {:conf {"username" "******", "password" "*******", "maxCommandsInProcess" 40}}```

I redacted the username and password, but they appear in the logs.

**To Reproduce**
Create a client where you supply a value greater than 0 for `maxCommandsInProcess` and don't supply `maxCommandsInQueue`.
 
**Expected behavior**
The username and password should be redacted as in the example above.

### PR DESCRIPTION
**Pull request by yaronel**

### CODE REVIEW

File: test/aerospike_clj/client_test.clj [hunk 1]
```diff
@@ -36,8 +37,24 @@
...
     (is c)
     (is (= "localhost" (:cluster-name c))))
-  (is (thrown-with-msg? Exception #"setting maxCommandsInProcess>0 and maxCommandsInQueue=0 creates an unbounded delay queue"
-                        (client/init-simple-aerospike-client ["localhost"] "test" {"maxCommandsInProcess" 1}))))
+
+  (letfn [(no-password? [ex]
+            (let [conf (:conf (ex-data ex))]
+              (and conf (not (contains? conf "password")))))]
+   (try
+     (client/init-simple-aerospike-client ["localhost"] "test" {"maxCommandsInProcess" 1})
```
Comment (barkanido) [inline test/aerospike_clj/client_test.clj]: can you use `(is thrown? ...`

Comment (yaronel) [reply test/aerospike_clj/client_test.clj]: I don’t see how. I want to examine the exception.

Comment (barkanido) [reply test/aerospike_clj/client_test.clj]: you can do `(is (thrown-with-msg? c re ...))` where c is the exception class and re is a regular expression. e.g.
```
(is (thrown-with-msg? ArithmeticException #"Divide by zero"
                      (/ 1 0)))
```

Comment (yaronel) [reply test/aerospike_clj/client_test.clj]: But I want to verify the exception data doesn't contain the password.

Comment (barkanido) [reply test/aerospike_clj/client_test.clj]: I might be missing something, nut I assume a strict regex will do (`#"^Devide by zero$"` is an example)

Comment (yaronel) [reply test/aerospike_clj/client_test.clj]: OK, I forgot `(is (thrown-with-msg?` returns the exception.
Fixed.

**Comment by yaronel** [comment]
I pushed another small fix to this MR.
The mock client's `healthy?` function was missing the one argument arity.

**Comment by barkanido** [comment]
Can you please merge latest master?

**Comment by yaronel** [comment]
Merged

File: src/aerospike_clj/client.clj [hunk 2]
```diff
@@ -4,7 +4,8 @@
             [aerospike-clj.utils :as utils]
             [aerospike-clj.metrics :as metrics]
             [aerospike-clj.key :as as-key]
-            [manifold.deferred :as d])
+            [manifold.deferred :as d]
+            [taoensso.timbre :as log])
```
Comment (barkanido) [inline src/aerospike_clj/client.clj]: timbre is not a production dependency (only testing)

**Review by barkanido — CHANGES_REQUESTED** [review]

**Review by barkanido — APPROVED** [review]

**Pull request merged**